File: ssr-simple/src/main/java/com/cincinnatiai/ssr_simple/model/NodeModel.kt
package com.cincinnatiai.ssr_simple.model

data class NodeModel(
    val type: String,
    val title: String? = null,
    val description: String? = null,
    val label: String? = null,
    val backgroundColor: String? = null,
    val elevation: Float? = null,
    val imageUrl: String? = null,
    val imageHeight: Float? = null,
    val imageWidth: Float? = null,
    val contentDescription: String? = null,
    val action: String? = null,
    val modifier: ModifierModel? = null,
    val textStyle: TextStyleModel? = null,
    val topBar: NodeModel? = null,
    val floatingActionButton: NodeModel? = null,
    val content: NodeModel? = null,
    val children: List<NodeModel>? = null,
    // Button-specific properties
    val buttonVariant: String? = null, // "filled", "outlined", "text" (default: "filled")
    // Table-specific properties
    val columns: List<TableColumnModel>? = null,
    val tableData: List<List<TableCellModel>>? = null,
    val showBorders: Boolean? = null,
    val headerBackgroundColor: String? = null,
    val rowAction: String? = null, // Action when entire row is clicked
    val roundedCorners: Float? = null, // Rounded corner radius in dp for tables
    val useLazyColumn: Boolean? = null // Use LazyColumn for table rows (default: true). Set to false when table is in a scrollable parent
)

File: ssr-simple/ssr-simple-react/src/models/NodeModel.ts
import { ModifierModel } from './ModifierModel';
import { TextStyleModel } from './TextStyleModel';
import { TableColumnModel } from './TableColumnModel';
import { TableCellModel } from './TableCellModel';

export interface NodeModel {
  type: string;
  title?: string;
  description?: string;
  label?: string;
  backgroundColor?: string;
  elevation?: number;
  imageUrl?: string;
  imageHeight?: number;
  imageWidth?: number;
  contentDescription?: string;
  action?: string;
  modifier?: ModifierModel;
  textStyle?: TextStyleModel;
  topBar?: NodeModel;
  floatingActionButton?: NodeModel;
  content?: NodeModel;
  children?: NodeModel[];

  // Button-specific properties
  buttonVariant?: 'filled' | 'outlined' | 'text';

  // Table-specific properties
  columns?: TableColumnModel[];
  tableData?: TableCellModel[][];
  showBorders?: boolean;
  headerBackgroundColor?: string;
  rowAction?: string; // Action when entire row is clicked
  roundedCorners?: number; // Rounded corner radius in px for tables
  useLazyColumn?: boolean; // Use virtualization for table rows (default: true)
}
